bolt-design-system/bolt
# Hotfix: Lang-related Sass and Build-related Fixes

## Jira
N/A

## Summary
Fixes a styling regression introduced with the original Bolt v2.0.0 release that prevents the default `@bolt/core` spacing scale values from being updated (ex. when adjusting the typography and spacing for specific languages like Japanese).

## Details
This also includes several Webpack and docs site-related fixes to account for the `lang` config in the `.boltrc` config not being set to any value, being set to a fixed value, being set to an array with one value, and being set to an array of multiple values. 

This includes issues seen on the docs site where no CSS / JS file would loaded when the config was set to a fixed string + a separate issue where the site would infinitely reload when attempting to run locally and visit the homepage (webpackHotMiddleware issue)

## How to test
1. Confirm a `.boltrc` config with the `lang` config set to `ja` visually is fixed on the demo Japanese page in Pattern Lab
2. Test to confirm the following all lang config options all  and work as expected -- both as a standalone front-end build AND with the local front dev server

> Note: I'd recommend clearing the `www/build` folder every time to make sure the files generated are what we expect

### Expected outcomes:
1. lang-specific file suffixes should only be applied when a lang config is specified AND the site is built in prod mode OR is run locally with the `--i18n` flag
2. lang-specific file suffixes should not be applied by default when a lang config isn't set
3. HMR (hot module replacement) should work for local development when no lang config or 1 lang config (in an array or string) is set
4. Full page reloads should work when changing JS files while doing local dev when 2 lang configs are set
5. The docs site shouldn't reload infinitely when 2 lang configs are set and running for local development
6. The doc site's CSS and JS should load as expected in local dev mode + building for prod without having to change any config files or paths, regardless on which lang config setting is set.
7. Any config with `ja` set -- either as a string, as the only value in an array, or as the **first** value in an array with multiple langs set -- should all render this page in Pattern Lab locally with smaller fonts and tighter spacing: https://boltdesignsystem.com/pattern-lab/?p=pages-homepage-japanese

#### 1. No `lang` config set in the `.boltrc` config file
- [x] Does the site run and visually load as expected when running `yarn run start`
- [x] Does the site run and visually load as expected when running `yarn run start --i18n`
- [x] Does the site fully compile compile when running `yarn run build`

#### 2a. Setting `lang` as a string (ie. `lang: 'ja',`)
- [x] is the Japanese typography fix visible with this config?
- [x] Does the site run and visually load as expected when running `yarn run start`
- [x] Does the site run and visually load as expected when running `yarn run start --i18n`
- [x] Does the site fully compile compile when running `yarn run build`

#### 2b. Setting `lang` as a string (ie. `lang: 'en',`)
- [x] Does the site run and visually load as expected when running `yarn run start`
- [x] Does the site run and visually load as expected when running `yarn run start --i18n`
- [ ] Does the site fully compile compile when running `yarn run build`

#### 3a. Setting `lang` as an array with one value (ie. `lang: ['en'],`)
- [x] Does the site run and visually load as expected when running `yarn run start`
- [x] Does the site run and visually load as expected when running `yarn run start --i18n`
- [ ] Does the site fully compile compile when running `yarn run build`

#### 3b. Setting `lang` as an array with one value (ie. `lang: ['ja'],`)
- [x] is the Japanese typography fix visible with this config?
- [x] Does the site run and visually load as expected when running `yarn run start`
- [x] Does the site run and visually load as expected when running `yarn run start --i18n`
- [ ] Does the site fully compile compile when running `yarn run build`

#### 4a. Setting `lang` as an array with two values (ie. `lang: ['en', 'ja'],`)
- [x] Does the site run and visually load as expected when running `yarn run start`
- [x] Does the site run and visually load as expected when running `yarn run start --i18n`
- [ ] Does the site fully compile compile when running `yarn run build`

#### 4b. Setting `lang` as an array with two values (ie. `lang: ['ja', 'en'],`)
- [x] is the Japanese typography fix visible with this config?
- [x] Does the site run and visually load as expected when running `yarn run start`
- [x] Does the site run and visually load as expected when running `yarn run start --i18n`
- [ ] Does the site fully compile compile when running `yarn run build`
